db/document.go

```diff
@@ -814,15 +814,7 @@ func (doc *Document) IsChannelRemoval(revID string) (bodyBytes []byte, history R
 	}
 	// If no matches found, return isRemoval=false
 	if len(removedChannels) == 0 {
-		return nil, nil, nil, false, false, nil
-	}
-
-	// Construct removal body
-	// doc ID and rev ID aren't required to be inserted here, as both of those are available in the request.
-	if isDelete {
```

Comment:
As discussed, returning bodyBytes=nil is going to cause subsequent calls to the rev cache to treat this as a cache miss.  So I think we still want to set some kind of body (to avoid a lot of 'is removed' checking every time someone tries to read something from the cache.  It still makes sense to set the 'removed' flag on the rev cache entry - it's a more efficient check that the body comparison we were previously doing - but in the case of a naive caller that neglects to check docRev.removed, we should return RemovedRedactedDocument.

Reply:
Makes sense, return  {“_removed”: true} from IsChannelRemoval irrespective of whether the removal was a tombstone.